File: Fahrenheit/src/com/zakharkin/example/lollipop/FahrenheitWindow.java
package com.zakharkin.example.lollipop;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Looper;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.util.HashMap;
import java.util.Map;

public class FahrenheitWindow extends Activity {
    private GeneralTranslator translator = new GeneralTranslator();
    Button buttonCelsius, buttonFahrenheit;
    Button buttonMmHg, buttonPsi;
    EditText editTextFrom;
    TextView resultText, textViewResult;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);
        buttonCelsius = (Button) findViewById(R.id.buttonCelsius);
        buttonFahrenheit = (Button) findViewById(R.id.buttonFahrenheit);
        buttonMmHg = (Button) findViewById(R.id.buttonMmHg);
        buttonPsi = (Button) findViewById(R.id.buttonPsi);
        editTextFrom = (EditText) findViewById(R.id.editTextFrom);
        textViewResult = (TextView) findViewById(R.id.textViewResultValue);
        resultText = (TextView) findViewById(R.id.textViewResult);
    }

    public void buttonCelsiusOnClick(View view) {
        translate(true);
        resultText.setText(buttonCelsius.getText());
    }

    public void buttonFahrenheitOnClick(View view){
        translate(false);
        resultText.setText(buttonFahrenheit.getText());
    }

    public void buttonMmHgOnClick(View view){
        pressTranslate(true);
        resultText.setText(buttonMmHg.getText());
    }

    public void buttonPsiOnClick(View view){
        pressTranslate(false);
        resultText.setText(buttonPsi.getText());
    }

    public void buttonViewOnClick(View view){
        Intent intent = new Intent(FahrenheitWindow.this, DBViewer.class);
        startActivity(intent);
    }

    private void translate(boolean toCelsius) {
        double from;
        try {
            from = Double.parseDouble(editTextFrom.getText().toString());
        }
        catch (Exception e){
            from = 0.0;
        }
        editTextFrom.setText(Double.toString(from));
        textViewResult.setText(Double.toString(toCelsius ? translator.ToCelsius(from) : translator.ToFahrenheit(from)));
    }

    private void pressTranslate(boolean toMmHg){
        double from;
        try{
            from = Double.parseDouble(editTextFrom.getText().toString());
        }
        catch (Exception e){
            from = 0.0;
        }
        editTextFrom.setText(Double.toString(from));
        textViewResult.setText(Double.toString(toMmHg ? translator.ToMmHg(from) : translator.ToPsi(from)));
    }
}

File: Fahrenheit/src/com/zakharkin/example/lollipop/FahrenheitWindow.kt
package com.zakharkin.example.lollipop

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.Looper
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

import java.util.HashMap

class FahrenheitWindow : Activity() {
    private val translator = GeneralTranslator()
    internal var buttonCelsius: Button? = null
    internal var buttonFahrenheit: Button? = null
    internal var buttonMmHg: Button? = null
    internal var buttonPsi: Button? = null
    internal var editTextFrom: EditText? = null
    internal var resultText: TextView? = null
    internal var textViewResult: TextView? = null

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)
        buttonCelsius = findViewById(R.id.buttonCelsius) as Button
        buttonFahrenheit = findViewById(R.id.buttonFahrenheit) as Button
        buttonMmHg = findViewById(R.id.buttonMmHg) as Button
        buttonPsi = findViewById(R.id.buttonPsi) as Button
        editTextFrom = findViewById(R.id.editTextFrom) as EditText
        textViewResult = findViewById(R.id.textViewResultValue) as TextView
        resultText = findViewById(R.id.textViewResult) as TextView
    }

    fun buttonCelsiusOnClick(view: View) {
        translate(true)
        resultText!!.text = buttonCelsius!!.text
    }

    fun buttonFahrenheitOnClick(view: View) {
        translate(false)
        resultText!!.text = buttonFahrenheit!!.text
    }

    fun buttonMmHgOnClick(view: View) {
        pressTranslate(true)
        resultText!!.text = buttonMmHg!!.text
    }

    fun buttonPsiOnClick(view: View) {
        pressTranslate(false)
        resultText!!.text = buttonPsi!!.text
    }

    fun buttonViewOnClick(view: View) {
        val intent = Intent(this@FahrenheitWindow, DBViewer::class.java)
        startActivity(intent)
    }

    private fun translate(toCelsius: Boolean) {
        val from: Double
        try {
            from = java.lang.Double.parseDouble(editTextFrom!!.text.toString())
        } catch (e: Exception) {
            from = 0.0
        }

        editTextFrom!!.setText(java.lang.Double.toString(from))
        textViewResult!!.text = java.lang.Double.toString(if (toCelsius) translator.ToCelsius(from) else translator.ToFahrenheit(from))
    }

    private fun pressTranslate(toMmHg: Boolean) {
        val from: Double
        try {
            from = java.lang.Double.parseDouble(editTextFrom!!.text.toString())
        } catch (e: Exception) {
            from = 0.0
        }

        editTextFrom!!.setText(java.lang.Double.toString(from))
        textViewResult!!.text = java.lang.Double.toString(if (toMmHg) translator.ToMmHg(from) else translator.ToPsi(from))
    }
}
